pvl/abridged/graphs/mst/prim.cpp

```diff
@@ -1,15 +1,20 @@
 #include "graph_template_adjlist.cpp"
 // insert inside graph; needs n, vis[], and adj[]
 void prim(viii &res, int s=0) {
-  viii().swap(res); // or use res.clear();
-  std::priority_queue<ii, vii, std::greater<ii> > pq;
-  pq.push{{0, s}};
+  res.clear();
+  std::priority_queue<iii, viii, std::greater<iii>> pq;
   vis[s] = true;
+  for (auto &[v, w] : adj[s])
+    if (!vis[v])
+      pq.push({w, {s, v}});
   while (!pq.empty()) {
-    int u = pq.top().second;    pq.pop();
+    auto edge = pq.top();    pq.pop();
+    int u = edge.second.second;
+    if (vis[u]) continue;
     vis[u] = true;
-    for (auto &[v, w] : adj[u]) {
-      if (v == u)   continue;
-      if (vis[v])   continue;
-      res.push_back({w, {u, v}});
-      pq.push({w, v}); } } }
+    res.push_back(edge);
+    for (auto &[v, w] : adj[u])
+      if (!vis[v])
+        pq.push({w, {u, v}});
+  }
+}
```

Comment:
[nit] compress into a single line

we use the stuff in `pvl/abridged/` to generate the team notebooks so they need to be as short as possible